aten/src/ATen/THLongStorageView.h

```diff
@@ -53,22 +53,30 @@ class THLongStorageView {
         break;
     }
 
-    if(zero_dim_to_one && ref.size() == 0) {
+    if (zero_dim_to_one && ref.size() == 0) {
       // make storage of size 0 actually a 1-length storage with 1 element
       // so that our 0-dim tensors get allocated as 1-dim inside TH
+
       one = 1;
-      storage.data_ptr = {&one, kCPU}; // non-owning
-      storage.size = 1;
+      storage = new StorageImpl(
+          at::CTypeToScalarType<th::from_type<int64_t>>::to(),
```

Comment:
I think the `th::from_type` here is not needed.

Reply:
I'll keep it since we'll delete this whole thing anyway.